### PR TITLE
Generate EPUB files

### PR DESCRIPTION
**Pull request by philschatz**
Refs https://github.com/openstax/ce/issues/1966

### CODE REVIEW

**Review by ckline-tryptic — APPROVED** [review]

File: bakery-js/src/minidom.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+  set attrs(attrs: Attrs) {
+    Object.keys(this.attrs)
+      .filter((v) => !Object.keys(attrs).includes(v))
+      .forEach((name) => this.el.removeAttribute(name))
+    Object.entries(attrs).forEach(([name, value]) =>
+      this.attr(name, value as string)
+    )
+  }
+  get attrs(): Attrs {
+    return Array.from(this.el.attributes).reduce(
```
Comment (TylerZeroMaster) [inline bakery-js/src/minidom.ts:85]: I think Object.fromEntries performs significantly better and is more readable:

```javascript
Object.fromEntries(Array.from(this.el.attributes).map(attr => [attr.name, attr.value]))
```

File: bakery-js/src/minidom.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+  setPos(doc.documentElement, source)
+
+  const $doc = dom(doc)
+  const $root = dom($doc.doc.documentElement)
+  $root.attrs = attrs
+  $root.children =
+    children === undefined
+      ? []
+      : Array.isArray(children)
+      ? children.map((c) => $doc.fromJSX(c))
```
Comment (TylerZeroMaster) [inline bakery-js/src/minidom.ts:334]: This could probably use additional indentation.

**Review by TylerZeroMaster — COMMENTED** [review]
I'm still working my way through this, but here's what I have found so far.

File: bakery-js/src/minidom.ts (hunk 2 above)
Comment (philschatz) [reply bakery-js/src/minidom.ts:334]: Yep, after debating how to format the code we opted to just use a code formatter. This reminded me that I need to set it up to run in CI (and maybe as a commit hook)

**Review by sparksam — APPROVED** [review]
Looks good to me.

**Review by therealmarv — APPROVED** [review]

**Pull request merged**